ui/src/app/kvm/components/KVMHeaderForms/KVMHeaderForms.tsx

```diff
@@ -20,35 +22,29 @@ const getFormComponent = (
   setHeaderContent: KVMSetHeaderContent,
   clearHeaderContent: ClearHeaderContent
 ) => {
-  // This is a reliable of differentiating a machine action from a pod action,
-  // but we should eventually try to have a consistent shape between them.
-  // https://github.com/canonical-web-and-design/maas-ui/issues/3017
-  if (
-    headerContent &&
-    typeof headerContent === "object" &&
-    "name" in headerContent
-  ) {
-    return (
-      <MachineActionForms
-        headerContent={headerContent}
-        setHeaderContent={setHeaderContent}
-      />
-    );
-  }
-
-  switch (headerContent) {
-    case PodAction.COMPOSE:
-      return <ComposeForm clearHeaderContent={clearHeaderContent} />;
-    case PodAction.DELETE:
-      return <DeleteForm clearHeaderContent={clearHeaderContent} />;
-    case PodAction.REFRESH:
-      return <RefreshForm clearHeaderContent={clearHeaderContent} />;
-    default:
-      return null;
+  if (isKVMOnlyHeaderContent(headerContent)) {
```

Comment:
Same question here about whether you need this util or just rely on the unique action names?

Reply:
So I needed it to coerce the type of `headerContent` to be either `KVMOnlyHeaderContent` or `MachineHeaderContent`, because the type required for the machine action title util is `MachineHeaderContent`. But it turns out TypeScript is smart enough to know that if all the cases of `KVMHeaderContent` are accounted for in the switch statement, then it knows that the `default` case will *have* to be of type `MachineHeaderContent`. So it turns out it's not necessary. And also that's amazing!